src/actions/Entities/NameNoteActions.js

```diff
@@ -63,7 +63,12 @@ const updateForm = (data, validator) => ({
 const saveEditing = () => (dispatch, getState) => {
   const nameNote = selectCreatingNameNote(getState()) ?? {};
   const patient = UIDatabase.get('Name', nameNote?.nameID);
-  const isDirty = JSON.stringify(patient?.mostRecentPCD?.data) !== JSON.stringify(nameNote?.data);
+  const isDirty =
+    !(
+      nameNote?.data &&
+      Object.keys(nameNote?.data).length === 0 &&
+      Object.getPrototypeOf(nameNote?.data) === Object.prototype
+    ) && JSON.stringify(patient?.mostRecentPCD?.data) !== JSON.stringify(nameNote?.data);
```

Comment:
I struggle to read this piece of code, could you maybe explain a little bit what's going on?
From what I can see, the `isDirty` variable basically means that the nameNote was updated.
The condition that you added here doesn't seem clear to me...

Reply:
The condition just check if the object is not empty is not empty, not null and is actually an object. 

The dirty variable was getting set to true when both the comaprion variable were empty ({}). 

Maybe we need to create utility function for this check, because this is for the case we are not even sure an object is sent in the variable or that object is empty or null and I see cases in our code where it can be useful. This can avoid app crashing.

Reply:
Thank you for the explanation, however, I can't understand why is there a `!` in front of it then:
```
!(
      nameNote?.data &&
      Object.keys(nameNote?.data).length === 0 &&
      Object.getPrototypeOf(nameNote?.data) === Object.prototype
    )
```

Reply:
If the object is empty, which this condition without the bang return as true, then the data is blank and is not dirty. So it has to not empty to go to the right hand side part of the conditional. It is an not empty check. a6ce058 this might make it more clear.